### PR TITLE
Downgrade android support libs to v23

### PR DESCRIPTION
**Pull request by zugaldia**
Companion PR to https://github.com/mapbox/mapbox-gl-native/pull/9698.

### CODE REVIEW

**Comment by LornTao** [comment]
@zugaldia @tobrun 
I have finished the code, and compile success in local. There are three steps needed to be follow-up.
- fixed the CI test issue.
- run the uploadArchives script in the module libandroid-telemetry, then get the public url,version info.
- change the dependencies of libandroid-telemetry in the [PR](https://github.com/mapbox/mapbox-gl-native/pull/9742).

**Comment by cammace** [comment]
Hey @LornTao, I rebased your branch so the `circle.yml` file would be moved into your branch and CI could pass.

**Comment by LornTao** [comment]
@cammace 
CI test failed because the gradle version is v2.14.1, log is as following:
```
cd mapbox; ./gradlew :libjava-core:assemble
Downloading https://services.gradle.org/distributions/gradle-2.14.1-all.zip
...........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
Unzipping /root/.gradle/wrapper/dists/gradle-2.14.1-all/8bnwg5hd3w55iofp58khbp6yv/gradle-2.14.1-all.zip to /root/.gradle/wrapper/dists/gradle-2.14.1-all/8bnwg5hd3w55iofp58khbp6yv
Set executable permissions for: /root/.gradle/wrapper/dists/gradle-2.14.1-all/8bnwg5hd3w55iofp58khbp6yv/gradle-2.14.1/bin/gradle

FAILURE: Build failed with an exception.

* What went wrong:
Could not open terminal for stdout: $TERM not set

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
Makefile:19: recipe for target 'build-release-java' failed
make: *** [build-release-java] Error 1
Exited with code 2
```

**Comment by zugaldia** [comment]
@LornTao That Gradle version is the one that [you're manually selecting](https://github.com/mapbox/mapbox-java/pull/534/files#diff-070d3a0bac61723adcfccdd2649a33e0R6). Why is that making the build fail? What's the exact error? What are your strategies to fix it?

File: mapbox/dependencies.gradle [hunk 1]
```diff
@@ -4,17 +4,17 @@ ext {
...
 
     dep = [
             // Support
-            supportAppcompat     : 'com.android.support:appcompat-v7:25.1.0',
-            supportCompat        : 'com.android.support:support-compat:25.1.0',
-            supportDesign        : 'com.android.support:design:25.1.0',
-            supportRecyclerview  : 'com.android.support:recyclerview-v7:25.1.0',
-            supportCardView      : 'com.android.support:cardview-v7:25.1.0',
+            supportAppcompat     : 'com.android.support:appcompat-v7:23.4.0',
+//            supportCompat        : 'com.android.support:support-compat:23.4.0',
```
Comment (zugaldia) [inline mapbox/dependencies.gradle]: @LornTao you don't need to comment out this line, it's just a version definition.

File: mapbox/libandroid-services/build.gradle [hunk 2]
```diff
@@ -37,7 +37,7 @@ dependencies {
 
     // Android Support libraries
     compile rootProject.ext.dep.supportAppcompat
-    compile rootProject.ext.dep.supportCompat
+//    compile rootProject.ext.dep.supportCompat
```
Comment (zugaldia) [inline mapbox/libandroid-services/build.gradle]: @LornTao don't you need to do the same for `libandroid-ui`?

File: mapbox/libandroid-telemetry/build.gradle [hunk 3]
```diff
@@ -2,7 +2,7 @@ apply plugin: 'com.android.library'
 
 android {
     compileSdkVersion rootProject.ext.compileSdkVersion
-    buildToolsVersion rootProject.ext.buildToolsVersion
+    buildToolsVersion '19.1.0'
```
Comment (zugaldia) [inline mapbox/libandroid-telemetry/build.gradle]: Could you change this directly in `dependencies.gradle`?

**Pull request closed without merging**